scripts: Makefile.lib: Add dependency for DTS overlay

agross-oss opened the pull request:
This patch adds a dependency for the DTS overlay so that the DTS is compiled
when the state of the overlay file changes.

Change-Id: I2affe67f90f56b1d97384d5cd4e3026abed24253
Signed-off-by: Andy Gross <andy.gross@linaro.org>

Commit:
scripts: Makefile.lib: Add dependency for DTS overlay

This patch adds a dependency for the DTS overlay so that the DTS is
compiled when the state of the overlay file changes.

Change-Id: I2affe67f90f56b1d97384d5cd4e3026abed24253
Signed-off-by: Andy Gross <andy.gross@linaro.org>

d3zd3z commented on scripts/Makefile.lib:
```diff
@@ -282,9 +282,15 @@ cmd_dt_S_dtb=						\
 $(obj)/%.dtb.S: $(obj)/%.dtb
 	$(call cmd,dt_S_dtb)
 
+ifneq ($(wildcard $(PROJECT_BASE)/$(BOARD_NAME).overlay),"")
```
Do we need this test? The code below tests for the file's existence, so why not just set it? Perhaps

    DTS_OVERLAY ?= $(PROJECT_BASE)/$(BOARD_NAME).overlay

setting it weakly to make it easier to override.

agross-linaro replied:
hmmm i tried the wildcard, but not this.  let me retest and check

d3zd3z commented on scripts/Makefile.lib:
```diff
@@ -282,9 +282,15 @@ cmd_dt_S_dtb=						\
 cmd_dtc = echo '\#include "$(notdir $<)"' > dts/$(ARCH)/$(notdir $<)_pre_compiled ; \
-	if test -e $(PROJECT_BASE)/$(BOARD_NAME).overlay; then \
+	if test -e $(DTS_OVERLAY); then \
 		echo '\#include "$(BOARD_NAME).overlay"' >> dts/$(ARCH)/$(notdir $<)_pre_compiled ; \
```
We use DTS_OVERLAY for the name in the test, but the #include pulls it in differently, and would ignore a setting by the user.

Perhaps use $(abspath $(PROJECT_BASE)/$(BOARD_NAME).overlay at least somewhere. Maybe have DTS_OVERLAY be user settable, and then set an absolute one to that, and use it for the #include and the dependencies.

d3zd3z commented on scripts/Makefile.lib:
```diff
@@ -296,7 +302,7 @@ cmd_dtc = echo '\#include "$(notdir $<)"' > dts/$(ARCH)/$(notdir $<)_pre_compile
 	$(call if_changed_dep,dtc)
 
-$(obj)/%.dts_compiled: $(src)/%.dts FORCE
+$(obj)/%.dts_compiled: $(src)/%.dts $(DTS_OVERLAY) FORCE
```
Does it really matter to add a dependency on something that is FORCEd?

agross-linaro replied:
Yes.  The call to if_dep_changed will catch the dependency change if we also add in the overlay file.  Otherwise it won't ever run the cmd_dts.

d3zd3z committed:
dts: Allow override of dts overlay directory

Instead of requiring the dts overlay files to be at the top level
directory (where make is invoked), allow this default to be overridden
by setting DTS_OVERLAY_DIR.  This is a directory where the overlays
themselves (which are still named $(BOARD_NAME).overlay) will live.

Change-Id: Ie9796afbd27971650b7636a36149c0d1f8e2b9fb
Signed-off-by: David Brown <david.brown@linaro.org>

d3zd3z reviewed (approved):
Looks good to me.
